### PR TITLE
Integrate Pipes with scheduler & syscalls

### PR DESCRIPTION
**Pull request by ThomasMiz**

### CODE REVIEW

File: Userland/SampleCodeModule/shell.c [hunk 1]
```diff
@@ -163,6 +164,33 @@ void help(void) {
 }
 
 void time(void) {
```
Comment (ichayer) [inline Userland/SampleCodeModule/shell.c:166]: ?

Comment (ThomasMiz) [reply Userland/SampleCodeModule/shell.c:166]: :)

**Comment by ichayer** [comment]
Change the commits name please, im going mad :)

**Comment by ThomasMiz** [comment]
> Change the commits name please, im going mad :)

You're not my mum 😠

File: Kernel/resourceNamerADT.c [hunk 2]
```diff
@@ -104,21 +104,22 @@ int rnm_nameResource(TResourceNamer namer, void* resource, const char* name, con
     return 0;
 }
 
-int rnm_unnameResource(TResourceNamer namer, const char* name) {
+void* rnm_unnameResource(TResourceNamer namer, const char* name) {
     int c;
     int index = findResourceIndex(namer, name, &c);
+    void* resource = namer->buffer[index].resource;
 
     // If there is no such resource with that name, return 1.
```
Comment (camilaDiToro) [inline Kernel/resourceNamerADT.c:112]: This comment might be deleted

**Review by camilaDiToro — APPROVED** [review]

**Review by ichayer — APPROVED** [review]

**Pull request merged**